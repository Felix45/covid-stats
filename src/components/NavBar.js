import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchTitleThunk } from '../redux/slices/navbarSlice';

const NavBar = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchTitleThunk('World'));
  }, []);

  const { title } = useSelector((state) => state.title);

  return (
    <header>
      <nav className="d-flex justify-content-between p-3">
        <ul>
          <li>
            <NavLink to="/">
              <span className="fa fa-chevron-left fa-lg" />
            </NavLink>
          </li>
        </ul>
        <div>
          <span>
            <strong>{title}</strong>
          </span>
        </div>
        <div>
          <span className="fa fa-microphone fa-lg mx-3" />
          <span className="fa fa-cog fa-lg" />
        </div>
      </nav>
    </header>
  );
};

export default NavBar;
