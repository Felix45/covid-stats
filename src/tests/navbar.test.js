import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux';
import NavBar from '../components/NavBar';

describe('Test if navBar renders correctly', () => {
  test('Test if navBar renders correctly', () => {
    const Tree = renderer.create(
      <Provider store={store()}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(Tree).toMatchSnapshot();
  });
});
