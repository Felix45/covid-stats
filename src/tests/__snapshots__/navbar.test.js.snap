// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if navBar renders correctly Test if navBar renders correctly 1`] = `
<header>
  <nav
    className="d-flex justify-content-between p-3"
  >
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          role="button"
        >
          <span
            className="fa fa-chevron-left fa-lg"
          />
        </a>
      </li>
    </ul>
    <div>
      <span>
        <strong>
          World
        </strong>
      </span>
    </div>
    <div>
      <span
        className="fa fa-microphone fa-lg mx-3"
      />
      <span
        className="fa fa-cog fa-lg"
      />
    </div>
  </nav>
</header>
`;
